File: LeetCode/JavaScript/0011-Container-With-Most-Water.js
// 11. Container With Most Water
// https://leetcode.com/problems/container-with-most-water/description/

/**
 * @param {number[]} height
 * @return {number}
 */

var maxArea = function (height) {
  let left = 0,
    right = height.length - 1;
  let maxProduct = 0;

  while (left < right) {
    let tempProduct = Math.min(height[left], height[right]) * (right - left);
    if (maxProduct < tempProduct) maxProduct = tempProduct;
    if (height[left] < height[right]) left++;
    else right--;
  }

  return maxProduct;
};

File: LeetCode/Python/0011_Container_With_Most_Water.py
# 11. Container With Most Water
# https://leetcode.com/problems/container-with-most-water/

class Solution:
    def maxArea(self, height: list[int]) -> int:
        area = 0
        l, r = 0, len(height) - 1

        while l <= r:
            temp = min(height[l], height[r]) * (r - l)

            if temp > area:
                area = temp

            if height[l] > height[r]:
                r -= 1
            else:
                l += 1

        return area
